bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/Ephemeris.scala

```diff
@@ -1,9 +1,46 @@
 package edu.gemini.spModel.core
 
-import scalaz._, Scalaz._
+import java.util.TimerTask
+import java.util.logging.Logger
 
-final case class Ephemeris(site: Site, data: Long ==>> Coordinates) {
+import scalaz._, Scalaz._
 
+final class Ephemeris(val site: Site, val compressedData: Deflated[List[(Long, Float, Float)]]) extends Serializable {
+
+  // Internal state for managing the inflated ephemeris data.
+  @transient @volatile private var _data: Long ==>> Coordinates = null
+  @transient @volatile private var _task: TimerTask = null
+
+  /**
+   * A map from time to coordinates. This value is stored in compressed form and is deflated (and
+   * cached for a while) on demand. The deflated value is transient, and is thus not serialized.
+   */
+  def data: Long ==>> Coordinates = synchronized {
+    import Ephemeris.{ timer, logger }
+
+    // Decompress if needed
+    if (_data == null) {
+      logger.info("Inflating compressed ephemeris.")
+      _data = ==>>.fromList(compressedData.inflate.map { case (t, r, d) =>
+        t -> Coordinates.fromDegrees(r, d).getOrElse(sys.error(s"corrupted ephemeris data: $t $r $d"))
+      })
+    }
+
+    // Reset the cache timer to a point the future
+    if (_task != null) _task.cancel()
+    _task = new TimerTask {
+      def run(): Unit = {
+        logger.info("Discarding inflated ephemeris.")
+        Ephemeris.this.synchronized(_data = null)
+      }
+    }
```

Comment:
This is quite a clever idea, and one that probably wouldn't have occurred to me.  I wonder about the performance in large programs in the OT though.  The OT is insistent about running P2 checks on every node in the program (probably multiple times) whenever anything changes.  Since there must be a P2 check or two that look for coordinates this means we'll be constantly deflating ephemeris data, I'm guessing.  Something similar may happen for observation query functors for the ODB.  For example, when building up the mini-model for QPT.

We'll need memory enough for the original deflated ephemeris data as well as the compressed data during those times.  It is probably worth profiling memory use while editing a large program with ephemeris data.